src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceResultSet.java

```diff
@@ -36,10 +36,12 @@
 import java.util.TimeZone;
 import java.util.concurrent.atomic.AtomicBoolean;
 
+import static com.salesforce.cdp.queryservice.util.Messages.QUERY_EXCEPTION;
+
 @Slf4j
 public class QueryServiceResultSet implements ResultSet {
 
-    private List<Map<String, Object>> data;
+    private List<Object> data;
```

Comment:
won't it be a good idea to implement a new result set class instead of adding if{}... else everywhere ?